Refactoring/main

BoBoBaSs84 opened the pull request.

BoBoBaSs84 committed:
this is it for now ...

BoBoBaSs84 committed:
well ... a start

BoBoBaSs84 committed:
good for now ...

BoBoBaSs84 committed:
tests added

BoBoBaSs84 committed:
more interfaces ...

BoBoBaSs84 committed:
good for now ...

BoBoBaSs84 committed:
this is it ...

Commit:
good commit  ...

Commit:
tests added, fixes added

Commit:
fixes ...

Commit:
good for now ...

Commit:
well well well

Commit:
small fixes ...

Commit:
platform fix

Commit:
build props and projects fixes ...

Commit:
Move to Core Project ...

Commit:
a good start for transactions ...

Commit:
removing the player inventory

Commit:
Quiete some fixes ...

Commit:
this is it for now ...

Commit:
more tests added

Commit:
fixes ...

Commit:
a good new start

Commit:
fixes, cleanUp

Commit:
assembly ...

Commit:
Settings added, logger service added, unit tests added

Commit:
config stuff added & changed

Commit:
lots of changes ...

Commit:
lots of factories created ...
lots of tests ...
this completes the state stuff for now ...

BoBoBaSs84 committed:
Merge pull request #49 from BoBoBaSs84/DEV/PlayerState

That'll do!

Commit:
good for now ...

Commit:
fix

BoBoBaSs84 committed:
this is it ...

Commit:
changes:
- menus started
- extensions added & changed
- unit tests added
- resources added

Commit:
changes:
- resources added
- transaction modified

Commit:
assembly fix

Commit:
well good for now ..

Commit:
well well well

Commit:
good for now ...

Commit:
well stiil things todo...

Commit:
this is it, folks!

BoBoBaSs84 committed:
Merge pull request #52 from BoBoBaSs84/DEV/refactoring-drug-interface-drug-class

Dev/refactoring drug interface drug class

Commit:
this is it, folks!

Closes #53

BoBoBaSs84 committed:
Merge pull request #55 from BoBoBaSs84/DEV/refactoring-unit-tests-for-all-factories

this is it, folks!

BoBoBaSs84 committed:
Merge pull request #56 from BoBoBaSs84/DEV/refactoring

Dev/refactoring

Commit:
This is it, folks!

Closes #54

BoBoBaSs84 committed:
Merge pull request #58 from BoBoBaSs84/DEV/54-refactoring-transaction-factory

Dev/54 refactoring transaction factory

Commit:
This is it, folks!

Closes #53

BoBoBaSs84 committed:
Merge pull request #62 from BoBoBaSs84/DE/Refactoring/unit-tests-for-all-factories

This is it, folks!

Commit:
This is it, folks!

Closes #61

Commit:
Resolving merge conflicts ...

Commit:
That'll do!

BoBoBaSs84 committed:
Merge pull request #63 from BoBoBaSs84/DEV/Refactoring/randomize-extensions-drug

Dev/refactoring/randomize extensions drug

Commit:
small project fix ...

Commit:
this is it, folks!

Closes #64

BoBoBaSs84 committed:
Merge pull request #65 from BoBoBaSs84/DEV/Refactoring/64-exception-message-drug-add-remove

this is it, folks!

Commit:
Merge branch 'main' into DEV/Refactoring/Main

Commit:
Merge branch 'main' into DEV/Refactoring/Main

BoBoBaSs84 committed:
fixes ...

Commit:
more fixes ...

Commit:
good for now ...

BoBoBaSs84 committed:
this is it, folks!

BoBoBaSs84 committed:
Merge pull request #71 from BoBoBaSs84/DEV/Refactoring/Fix/70-assembly-and-projetcs

this is it, folks!

BoBoBaSs84 committed:
merge from main

BoBoBaSs84 committed:
This is it, folks!

BoBoBaSs84 committed:
now it's fine ...

BoBoBaSs84 committed:
This is it, folks!

Closes #73

BoBoBaSs84 committed:
Merge pull request #74 from BoBoBaSs84/DEV/Refactoring/Compression

Dev/refactoring/compression

BoBoBaSs84 committed:
merge from main

Commit:
stuff changed ...

Commit:
This will do the trick!

Closes #75

BoBoBaSs84 committed:
Merge pull request #76 from BoBoBaSs84/Refactoring/Feature/75-settings-menu

This will do the trick!

Commit:
Resolving them conflicts ...

Commit:
last commit for now ...

Commit:
this is it for now...

BoBoBaSs84 committed:
Merge pull request #77 from BoBoBaSs84/DEV/Refactoring/Feature/69-the-menu

Dev/refactoring/feature/69 the menu

Commit:
Whoop Whoop!

Changes:
- fixed transaction unit test

Commit:
mh... let's test that ...

Commit:
This is it, folks!

Closes #78

BoBoBaSs84 committed:
Merge pull request #81 from BoBoBaSs84/Bug/Refactoring/78-the-price-comparison-doesnt-work-correctly

Bug/refactoring/78 the price comparison doesnt work correctly

Commit:
not bad, not bad ...

Commit:
goood for now ...

Commit:
getting better ...

Commit:
well well well ...
tests and menu next ...

Commit:
well good i think ...

Commit:
this is it ...

Commit:
this is it for now ...

Commit:
This is it, folks!

Now ... testing ...

BoBoBaSs84 committed:
Merge pull request #83 from BoBoBaSs84/Refactoring/Feature/80-add-more-settings

Refactoring/feature/80 add more settings

Commit:
feat: add probability to drug attribute

Changes:
- drug attribute
- drug attribute extensions
- unit tests
- drug class

Closes #82

BoBoBaSs84 committed:
Merge pull request #84 from BoBoBaSs84/Feature/Refactoring/82-drugattribute

feat: add probability to drug attribute

Commit:
ein juter zwischenstand ...

Commit:
ein juter zwischen stand ...

Commit:
feat: Game state service

Changes:
- a lot!

Closes #85

BoBoBaSs84 committed:
Merge pull request #88 from BoBoBaSs84/Refactoring/Feature/85-game-state-service

Refactoring/feature/85 game state service

BoBoBaSs84 committed:
feat: Player interface and the side menu

changes:
- quite a lot
- helper creted
- unit tests created and changed

closes #87

BoBoBaSs84 committed:
Merge pull request #89 from BoBoBaSs84/Refactoring/Feature/87-player-interface-and-the-side-menu

feat: Player interface and the side menu

BoBoBaSs84 committed:
bad commit, very bad ...

BoBoBaSs84 committed:
feat: Transaction log aka. history

changes:
- a lot

closes #79

BoBoBaSs84 committed:
Merge pull request #90 from BoBoBaSs84/Refactoring/Feature/79-transaction-log-aka-history

feat: transaction log aka history

Commit:
fix: transaction log

changes:
- side menu
- transaction service interface
- transaction service
- transaction service unit tests

BoBoBaSs84 committed:
Merge pull request #92 from BoBoBaSs84/Refactoring/Fix/91-transaction-log

fix: transaction log

Commit:
fix: The pipe line

changes:
- removed old project

closes #72

BoBoBaSs84 committed:
Merge pull request #93 from BoBoBaSs84/BugFix/72-fix-the-pipe-line

fix: The pipe line

BoBoBaSs84 committed:
Update dotnet.yml

BoBoBaSs84 committed:
Update dotnet.yml

BoBoBaSs84 committed:
Merge pull request #94 from BoBoBaSs84/BoBoBaSs84-patch-1

Update dotnet.yml

BoBoBaSs84 committed:
Update dotnet.yml

BoBoBaSs84 committed:
Merge pull request #95 from BoBoBaSs84/BoBoBaSs84-patch-1

Update dotnet.yml

BoBoBaSs84 committed:
Update dotnet.yml

BoBoBaSs84 committed:
Merge pull request #96 from BoBoBaSs84/BoBoBaSs84-patch-1

Update dotnet.yml

Commit:
feat: Dealer creation with ped hash

changes:
- pedestrian class & interface
- dealer class & interface
- dealerstate & drustate
- div. factories
- unit tests

closes #97

BoBoBaSs84 committed:
Merge pull request #98 from BoBoBaSs84/Refactoring/Feature/97-dealer-creation-with-ped-hash

feat: Dealer creation with ped hash

Commit:
Good for now ....

Commit:
lots of changes ...
still in progress

Commit:
readme updated

Commit:
nuget update

BoBoBaSs84 committed:
Merge pull request #104 from BoBoBaSs84/refactoring/feature/99-trafficking-mission

Refactoring/feature/99 trafficking mission

Commit:
refactor: deleted old project and files

Commit:
feat: central package management

changes:
- Directory.Packages.props added
- project files modified

closes #105

Commit:
feat: clean architecture

changes:
- added domain layer (models, enumerators, ... )
- added infrastructure layer (logger, file system, ...)
- added abstractions layer(nterfaces)
- added presentation layer (Menus,menu items, ...)
- added application layer (scripts, missions, ... )

closes #106

Commit:
Merge remote-tracking branch 'origin/main' into DEV/Refactoring/Main

Commit:
good for now ...

Commit:
feat: interface for switch item

changes:
- only small changes
- making all menu items internal

BoBoBaSs84 committed:
feat: clean architecture tests

changes:
- model factory tests added
- actor factory tests added
- infrastructure factory tests added

BoBoBaSs84 committed:
good for now ...

BoBoBaSs84 committed:
well ... on going ...

BoBoBaSs84 committed:
more stuff added

BoBoBaSs84 committed:
feat: clean architecture tests

changes:
- LSDW project removed
- LSDW.Core project removed
- LSDW.Tests project removed
- LSDW.Core.Tests project removed
- unit tests added

closes #107

BoBoBaSs84 committed:
refactor: moving abstractions

BoBoBaSs84 committed:
feat: more overall abstraction

closes #108

BoBoBaSs84 committed:
Merge pull request #110 from BoBoBaSs84/Refactoring/107-clean-architecture-tests

Refactoring/107 clean architecture tests

BoBoBaSs84 committed:
feat: experience gain from transactions

changes:
- SideMenuHelper method added
- SideMenu modified
- tests added

#111

BoBoBaSs84 committed:
Merge pull request #112 from BoBoBaSs84/Refactoring/Feature/111-experience-gain-from-transactions

feat: experience gain from transactions

BoBoBaSs84 committed:
feat: extension methods for presentation layer

changes:
- switch item helper added
- switch item helper tests added

closes #113

BoBoBaSs84 committed:
Merge pull request #114 from BoBoBaSs84/Refactoring/Feature/113-extension-methods-for-presentation-layer

feat: extension methods for presentation layer

BoBoBaSs84 committed:
feat: trafficking mission

changes:
- settings added
- unit tests added

BoBoBaSs84 committed:
feat: Xml elements and attribute names

changes:
- state class modifed
- xml constants class modified
- DrugType parameter renamed

closes #115

BoBoBaSs84 committed:
Merge pull request #116 from BoBoBaSs84/Refactoring/Feature/115-xml-elements-and-attribute-names

feat: Xml elements and attribute names

BoBoBaSs84 committed:
merge from main

BoBoBaSs84 committed:
refactor: settings service to infrastructure

changes:
- total rework of the settings related things
- moved the service from domain to infrastructure
- moved the interface from domain to abstractions
- text templating for presentation added

... still some things left todo

BoBoBaSs84 committed:
feat: Settings service to infrastructure

changes:
- quite some ...
- that'll do it

closes #117

BoBoBaSs84 committed:
feat: trafficking Mission

changes:
- settings added
- unit tests added

BoBoBaSs84 committed:
Merge pull request #118 from BoBoBaSs84/Refactoring/Feature/99-trafficking-mission

Refactoring/feature/99 trafficking mission

BoBoBaSs84 committed:
good for now ...

BoBoBaSs84 committed:
docs: readme update

BoBoBaSs84 committed:
feat: trafficking Mission

changes:
- mission base class added
- mission base interface modified
- trafficking class changed

BoBoBaSs84 committed:
feat: trafficking mission

changes:
- price change interval setting added
- inventory change interval setting added

BoBoBaSs84 committed:
refactor: ScriptHookHelper

BoBoBaSs84 committed:
fix: Experience multiplier not used

changes:
- SideMenuHelper class method changed
- fixes in dealer actor class
- fixes in pdestrian actor class

closes #119

BoBoBaSs84 committed:
Merge pull request #120 from BoBoBaSs84/Refactoring/Fix/119-experience-multiplier-not-used

fix: Experience multiplier not used

BoBoBaSs84 committed:
Merge pull request #121 from BoBoBaSs84/Refactoring/Feature/99-trafficking-mission

Refactoring/feature/99 trafficking mission

BoBoBaSs84 committed:
Update README.md

BoBoBaSs84 committed:
feat: trafficking Mission

changes:
- IEnumerable<IDealer> to ICollection<IDealer>

BoBoBaSs84 committed:
fix: small fixes

BoBoBaSs84 committed:
feat: Experience gain from transaction

changes:
- player interface modified
- player class modified
- transaction extension added
- transaction types added
- domain factories modified

BoBoBaSs84 committed:
feat: Experience gain from transaction

changes:
- removed the menu type
- reworked the whole presentation layer
- modified the transaction type
- cleanup resources

closes #122

BoBoBaSs84 committed:
Merge pull request #123 from BoBoBaSs84/Refactoring/Feature/122-experience-gain-from-transaction

feat: Experience gain from transaction

BoBoBaSs84 committed:
merge from refactoring/main

BoBoBaSs84 committed:
test added

BoBoBaSs84 committed:
feat: Refactoring

changes:
- removed the total profit property

BoBoBaSs84 committed:
feat: Refactoring

changes:
- removed market value property
- added average price property
- added more comments
- changed extensions

BoBoBaSs84 committed:
feat: Refactoring

changes:
- IEnumerable<IDealer> to ICollection<IDealer>
- Drug fixes
- tests added
- dealer method added

BoBoBaSs84 committed:
merge from Refactoring/main

BoBoBaSs84 committed:
refactor: clean up the abstraction layer

BoBoBaSs84 committed:
refactor: clean up

BoBoBaSs84 committed:
refactor: even more clean up and renaming

BoBoBaSs84 committed:
merge from refactoring/main

BoBoBaSs84 committed:
refactor: clean up

BoBoBaSs84 committed:
feat: trafficking mission

changes:
- extension methods added
- unit tests added
- dealer class and interface modified

Commit:
feat: trafficking mission

changes:
- DealerExtensions modified
- InventoryExtensions modified
- unit tests added

closes #125

Commit:
feat: notification system

changes:
- NotificationService class added
- NotificationService interface added
- factory added

Closes #126

BoBoBaSs84 committed:
Merge pull request #127 from BoBoBaSs84/Refactoring/Feature/126-notification-system

feat: notification system

Commit:
feat: Notification service and transaction service

Changes:
- transaction service class, interface, factoriy

Closes #128

Commit:
feat: Moq and base test project

it is a start

Closes #129

BoBoBaSs84 committed:
Merge pull request #130 from BoBoBaSs84/Refactoring/UnitTests/129-moq-and-base-test-project

feat: Moq and base test project

Commit:
feat: trafficking mission

changes:
- last refresh property added
- last restock property added
- unit tests modified

Commit:
feat: Trafficking Mission

changes:
- small ones...

Commit:
feat: Trafficking Mission

Changes:
- added trafficking extensions
- modified trafficking class and interface

closes #131

Commit:
feat: trafficking mission

changes:
- quite some
- unit tests are not working

BoBoBaSs84 committed:
feat: trafficking mission

changes:
- a lot and even more ...

closes #132 & #133

Commit:
feat: trfficking mission

changes:
- tests added

BoBoBaSs84 committed:
feat: version bump

changes:
- projects

closes #134

BoBoBaSs84 committed:
Merge pull request #135 from BoBoBaSs84/NuGet/134-version-bump

feat: version bump

Probotus committed:
fix: trafficking missions

Probotus committed:
feat: trafficking mission

prod build ...

Probotus committed:
stuff added ...
ped craeting not working correct ...

Probotus committed:
release build ...

Commit:
feat: Location Provider and interface

changes:
a lot...

closes #136

Commit:
refactor: INotificationService -> INotificationProvider

changes:
- a lot ...

closes #137

BoBoBaSs84 committed:
Merge pull request #138 from BoBoBaSs84/Refactoring/Feature/136-location-provider-and-interface

Refactoring/feature/136 location provider and interface

Probotus committed:
feat: trafficking mission

changes:
- trafficking interface
- logger service
- providers
- dealer model

Probotus committed:
feat: trafficking mission

changes:
a lot....

Probotus committed:
feat: trafficking mission

changes:
- a lot ...
- unit tests added
- game state service -> state service
- dealer methods added
- provider do not have constructors anymore

BoBoBaSs84 committed:
Merge pull request #140 from BoBoBaSs84/Refactoring/Feature/99-trafficking-mission

Refactoring/feature/99 trafficking mission

Commit:
refactor: Version bump LemonUI

changes:
- nuget updated
- led into the need for refactoring
- version bump was breaking

closes #141

BoBoBaSs84 committed:
Merge pull request #142 from BoBoBaSs84/141-nuget-version-bump-lemonui

refactor: Version bump LemonUI

Probotus committed:
Merge pull request #143 from BoBoBaSs84/Refactoring/main

Refactoring/main

Probotus committed:
feat: central package management

changes:
- props and projects

closes #144

Probotus committed:
feat: trafficking mission

changes:
- some fixes
- loading now works
- removed some clean up

Probotus committed:
good for now ...
found one hell of a nasty bug ...

Probotus committed:
feat: trafficking mission

- stuff added

Probotus committed:
more stuff added.

Probotus committed:
need help for that ...

Commit:
fix: move missions to application layer

changes:
- moved every mission relevant stuff to the application layer

closes #146

BoBoBaSs84 committed:
Merge pull request #148 from BoBoBaSs84/Refactoring/Feature/99-trafficking-mission

fix: move missions to application layer

Commit:
this is it for now ....

closes #147

Probotus committed:
Merge pull request #149 from BoBoBaSs84/Refactoring/UnitTests/147-more-tests-for-the-dealer-model

Refactoring/unit tests/147 more tests for the dealer model

Probotus committed:
well, i have the feeling this is not working ...

Commit:
feat: trafficking mission

changes:
- too much to tell
- need to test this ...

Probotus committed:
good for now ...

Commit:
feat: Player provider and character provider

changes:
- player provider interface and implementation added
- factory added
- unit tests added

closes #150

BoBoBaSs84 committed:
Merge pull request #151 from BoBoBaSs84/Refactoring/Feature/150-player-provider-and-character-provider

feat: Player provider and character provider

Probotus committed:
feat: world provider

changes:
- world provider interface and implementation added
- renamings
- unit tests modified

Probotus committed:
feat: world provider

changes:
- removed time provider -> changed to world provider

closes #154

BoBoBaSs84 committed:
Merge pull request #156 from BoBoBaSs84/Refactoring/154-location-provider-to-world-provider

feat: world provider

BoBoBaSs84 committed:
fix: Any() vs. Count vs. Count()

changes:
- quite some

closes #157

Commit:
fix: settings text template

changes:
- moved the settings file from domain into abstractions

closes #155

BoBoBaSs84 committed:
Merge pull request #160 from BoBoBaSs84/Refactoring/Fix/155-settings-text-template

fix: settings text template

Probotus reviewed (changes requested):
This should not be merged, the main mission is not working.

Commit:
refactor: Use property setters if possible

changes:
- dealer and dealer interfaces changed
- dealer extension methods changed

Commit:
refactor: Use property setters if possible

changes:
- unit tests added

closes: #162

BoBoBaSs84 committed:
Merge pull request #163 from BoBoBaSs84/Refactoring/162-use-property-setters-if-possible

Refactoring/162 use property setters if possible

Commit:
Fix: add assemblyinfo.cs to .gitignore

changes:
gitignore file

Commit:
mh...

Commit:
fix

Commit:
test

Commit:
test2

BoBoBaSs84 committed:
Merge pull request #167 from BoBoBaSs84/Refactoring/Fix/166-add-assemblyinfocs-to-gitignore

Fix: add assemblyinfo.cs to .gitignore

Commit:
feat: audio provider interface and implementation

changes:
- Audio provider interface added
- Audio provider implementation
- factory added

BoBoBaSs84 committed:
Merge pull request #172 from BoBoBaSs84/Refactoring/Feature/171-audio-provider-interface-and-implementation

feat: audio provider interface and implementation
